torchrl/data/tensor_specs.py

```diff
@@ -1118,7 +1122,7 @@ def __repr__(self) -> str:
     def type_check(
         self,
         value: Union[torch.Tensor, TensorDictBase],
-        selected_keys: Union[str, Optional[Sequence[str]]] = None,
+        selected_keys: Optional[Union[str, Optional[Sequence[str]]]] = None,
```

Comment:
Maybe `Optional[Union[str, Optional[Sequence[str]]]]` can be shortened to `Optional[Union[str, Sequence[str]]]`